[SPARK-25170][DOC] Add list and short description of Spark Executor Task Metrics to the documentation

LucaCanali opened the pull request:
## What changes were proposed in this pull request?

Add description of Task Metrics to the documentation.

## How was this patch tested?

None.

LucaCanali committed:
Document Spark Executor Task Metrics

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

kiszk commented on docs/monitoring.md:
```diff
@@ -388,6 +388,163 @@ value triggering garbage collection on jobs, and `spark.ui.retainedStages` that
+  </tr>
+  <tr>
+    <td>executorRunTime</td>
+    <td>Time the executor spent running this task. This includes time fetching shuffle data.
```
Does `Time` mean `elapsed time` or other `time`?

kiszk commented on docs/monitoring.md:
```diff
@@ -388,6 +388,163 @@ value triggering garbage collection on jobs, and `spark.ui.retainedStages` that
+    The value is expressed in milliseconds.</td>
+  </tr>
+  <tr>
+    <td>executorCpuTime
```
Do we miss `<td>` in these two lines?

kiszk commented on docs/monitoring.md:
```diff
@@ -388,6 +388,163 @@ value triggering garbage collection on jobs, and `spark.ui.retainedStages` that
+  </tr>
+  <tr>
+    <td>jvmGCTime</td>
+    <td>Amount of time the JVM spent in garbage collection while executing this task.
```
Why do we start with `amount of` while the above parameters start `Time` or `CPU Time`?

kiszk commented on docs/monitoring.md:
```diff
@@ -388,6 +388,163 @@ value triggering garbage collection on jobs, and `spark.ui.retainedStages` that
+  </tr>
+  <tr>
+    <td>resultSerializationTime</td>
+    <td>Amount of time spent serializing the task result.
```
ditto

kiszk commented:
I like to add description for metrics.

LucaCanali commented:
Thanks @kiszk for reviewing this. I have addressed your comments in a new commit +
apologies as I have now moved this to a new PR https://github.com/apache/spark/pull/22397 
I am closing this to avoid confusion.
